### ISSUE TITLE
Falling back to Ansible unique filter as Jinja2 one failed: '<variable>' is undefined

### ISSUE THREAD

**Issue by Noksa** [ansible/ansible-lint#3216]
##### Summary

Ansible-lint complains about undefined variable as warning

##### Issue Type

- Bug Report

##### OS / ENVIRONMENT

```console (paste below)
ansible-lint --version
ansible-lint 6.14.3 using ansible 2.14.3
```

- ansible installation method: pip
- ansible-lint installation method: pip

##### STEPS TO REPRODUCE

Create playbook `debug.yml`
```console (paste below)
- name: Debug
  hosts: localhost
  gather_facts: false
  tasks:
    - name: Set fact
      ansible.builtin.set_fact:
        qq: ["qq", "ww"]
    - name: Print it
      ansible.builtin.debug:
        msg: "{{ qq | unique }}"
```
Run ansible-lint
```shell
ansible-lint debug.yml
```

##### Desired Behavior

Output should be like that
```
❯ debug (master) ✘ ansible-lint debug.yml

Passed with production profile: 0 failure(s), 0 warning(s) on 1 files.
```

##### Actual Behavior

I see warning in output
```shell
❯ debug (master) ✘ ansible-lint debug.yml                    
[WARNING]: Falling back to Ansible unique filter as Jinja2 one failed: 'qq' is undefined

Passed with production profile: 0 failure(s), 0 warning(s) on 1 files.
```

I tried to use `unique | list` and `unique | flatten(1)` - didn't help.

Is it possible to avoid such warning? I have 10+ places with `unique` filter and it is annoying a bit.

[minimum complete verifiable example]: http://stackoverflow.com/help/mcve

**Comment by ssbarnea** [comment]
One of the problems about this one is that this warning is generated by ansible-core and directly sent to stdout, bypassing us completely.

That is not a standard python logging or warnings implementation, it is just a plain print, so capturing it is very tricky. I have being doing some work/research for redirecting stdout and stderr during run but it proved to cause problems so far, especially with our pytest unitests.

We will sort it, but it might take few more weeks, for now you can ignore the message.

**Comment by cidrblock** [comment]
Looks like that filter has some history to it, and came out before the jinja filter.   It's error is here https://github.com/ansible/ansible/blob/devel/lib/ansible/plugins/filter/mathstuff.py#L68 which was implemted here https://github.com/ansible/ansible/pull/45637/files...

It's unfortunate that the warnings ansible emits don't have an ID to them, b/c we end up having to string match everywhere ugh

**Comment by bcoca** [comment]
@chriscroome 

> It's a shame the Ansible filter wasn't called something different, for example `uniq`, to avoid this man_shrugging .

Hard to do since the Ansible filter predates the Jinja one. We are looking for a way in core to add a 'Jinja namespace' to allow explicit use of one of the other.

In general the 'matching name filters' should deffer to the Jinja one once present, in some cases we make the decision based on input as the acceptable options are not always the same.

**Comment by chriscroome** [comment]
> the Ansible filter predates the Jinja one

Thnaks @bcoca, I wasn't aware of that when I posted the above. :+1:
